### PR TITLE
[#175119553] Centralize environment variables in single config module

### PR DESCRIPTION
**Pull request by infantesimone**
This PR introduces the utils/config module which aims to be the unique interface for application configuration.

Every place in which the application used to directly access env variable has been refactored to use this module instead.

### CODE REVIEW

**Review by AleDore — COMMENTED** [review]
lgtm. Moreover the title of the PR should be more descriptive ( something like "Centralize Env variables in single config module" )

File: GetMessage/index.ts [hunk 1]
```diff
@@ -18,18 +17,22 @@ import createAzureFunctionHandler from "io-functions-express/dist/src/createAzur
...
 
-const messageContainerName = getRequiredStringEnv("MESSAGE_CONTAINER_NAME");
+const config = getConfigOrThrow();
+
+const messageContainerName = config.MESSAGE_CONTAINER_NAME;
 
 const messageModel = new MessageModel(
   cosmosdbInstance.container(MESSAGE_COLLECTION_NAME),
   messageContainerName
 );
```
Comment (BurnedMarshal) [inline GetMessage/index.ts:33]: Can we avoid when possible constant aliasing?
```suggestion
const messageModel = new MessageModel(
  cosmosdbInstance.container(MESSAGE_COLLECTION_NAME),
  config.MESSAGE_CONTAINER_NAME
);
```

Comment (infantesimone) [reply GetMessage/index.ts:33]: Ok thanks

File: StoreSpidLogs/__test__/index.test.ts [hunk 2]
```diff
@@ -3,15 +3,32 @@
 
 process.env = {
   APPINSIGHTS_INSTRUMENTATIONKEY: "foo",
+  AZURE_NH_ENDPOINT: "azendpoint",
```
Comment (gunzip) [inline StoreSpidLogs/__test__/index.test.ts:6]: This is hard to maintain (you have to updated tests for every new env). I suggest another approach:

1. create an env.example like here: https://github.com/pagopa/io-functions-admin/blob/master/env.example
with all the variables (eventually: document them with a comment inside the env.example file)
1. make jest read the env file using this configuration: https://github.com/pagopa/io-functions-admin/blob/master/jest.config.js#L3
1. remove all references to process.env in tests files

**Review by gunzip — CHANGES_REQUESTED** [review]

**Comment by balanza** [comment]
lgtm besides of the use of env variables in unit tests. I see this is needed in some cases because test suites import the whole function's `index.ts` file, which contains the initialisation code (hence env config is expected) (see [here](https://github.com/pagopa/io-functions-app/blob/90b660e1cf44434ac28339d0ef2f63c06f632a9e/StoreSpidLogs/__test__/index.test.ts#L21)).
Or in other cases, the handler directly imports a module which has a required env variable (see [here](https://github.com/pagopa/io-functions-app/blob/b816917916b7127b7530b30c85b3b83a63579beb/HandleNHNotificationCallActivity/handler.ts#L18)).
 
I see those are legacy problems, maybe we can take advantage to solve them by refactor such functions. The scope is: to only test `handler.ts` modules and to not directly reference env var in them.

**Comment by gunzip** [comment]
> I see those are legacy problems, maybe we can take advantage to solve them by refactor such functions. The scope is: to only test handler.ts modules and to not directly reference env var in them.

that's right, we can do it with another PR

File: StoreSpidLogs/__test__/index.test.ts (hunk 2 above)
Comment (infantesimone) [reply StoreSpidLogs/__test__/index.test.ts]: I have created the env file for testing, 
and added dotenv 8.2.0 as a dependency, it's ok?

File: StoreSpidLogs/__test__/index.test.ts [hunk 3]
```diff
@@ -92,6 +79,7 @@ describe("StoreSpidLogs", () => {
       }
     };
     const blobItem = await index(mockedContext as any, aSpidMsgItem);
+    console.log(blobItem);
```
Comment (gunzip) [inline StoreSpidLogs/__test__/index.test.ts:82]: woops

File: package.json [hunk 4]
```diff
@@ -32,6 +32,7 @@
     "@types/nodemailer": "^4.6.8",
     "danger": "^4.0.2",
     "danger-plugin-digitalcitizenship": "^0.3.1",
+    "dotenv": "^8.2.0",
```
Comment (gunzip) [inline package.json:35]: a question, does dotenv configured in jest.config.js read `env.example` by default ?
I know it reads `.env` by default but maybe env.example is included as well.

Comment (balanza) [reply package.json:35]: I think it's not, we might specify it explicitly

**Comment by balanza** [comment]
@infantesimone I see there are lint issues. No problem, you can fix them and push again. Try https://github.com/pagopa/git-hooks to prevent yourself from pushing bad code and save a roundtrip ;)

> LGTM but fix lint errors before merging.

File: package.json (hunk 4 above)
Comment (gunzip) [reply package.json:35]: done :)

Comment (infantesimone) [reply package.json:35]: Yes, the correct file name is `.env`. 
There is a comment in the first line of this file that indicates to rename, the idea is to indicate that it is just a test file.

**Comment by gunzip** [comment]
lgtm, why still draft?

**Review by gunzip — APPROVED** [review]

**Pull request merged**